src/Exceptionless.Insulation/Bootstrapper.cs

```diff
@@ -26,162 +26,188 @@
+                case "graphite":
                     metricsBuilder.Report.ToGraphite(new MetricsReportingGraphiteOptions {
                         Graphite = {
-                            BaseUri = new Uri(graphiteConnectionString.ServerUrl)
+                            BaseUri = new Uri(options.Data.GetString("server"))
                         }
                     });
                     break;
                 default:
                     return null;
             }
+            
             return metricsBuilder.Build();
         }
 
-        private static void RegisterMetricsReporting(IServiceCollection container) {
-            var connectionString = Settings.Current.MetricsConnectionString;
-            if (connectionString is StatsDMetricsConnectionString statsdConnectionString) {
-                container.ReplaceSingleton<IMetricsClient>(s => new StatsDMetricsClient(new StatsDMetricsClientOptions {
-                    ServerName = statsdConnectionString.ServerName,
-                    Port = statsdConnectionString.ServerPort,
-                    Prefix = "ex",
+        private static void RegisterQueue(IServiceCollection container, QueueOptions options, bool runMaintenanceTasks) {
+            if (String.Equals(options.Provider, "redis")) {
+                container.ReplaceSingleton(s => CreateRedisQueue<EventPost>(s, options, runMaintenanceTasks, retries: 1));
+                container.ReplaceSingleton(s => CreateRedisQueue<EventUserDescription>(s, options, runMaintenanceTasks));
+                container.ReplaceSingleton(s => CreateRedisQueue<EventNotificationWorkItem>(s, options, runMaintenanceTasks));
+                container.ReplaceSingleton(s => CreateRedisQueue<WebHookNotification>(s, options, runMaintenanceTasks));
+                container.ReplaceSingleton(s => CreateRedisQueue<MailMessage>(s, options, runMaintenanceTasks));
+                container.ReplaceSingleton(s => CreateRedisQueue<WorkItemData>(s, options, runMaintenanceTasks, workItemTimeout: TimeSpan.FromHours(1)));
+            } else if (String.Equals(options.Provider, "azurestorage")) {
+                container.ReplaceSingleton(s => CreateAzureStorageQueue<EventPost>(s, options, retries: 1));
+                container.ReplaceSingleton(s => CreateAzureStorageQueue<EventUserDescription>(s, options));
+                container.ReplaceSingleton(s => CreateAzureStorageQueue<EventNotificationWorkItem>(s, options));
+                container.ReplaceSingleton(s => CreateAzureStorageQueue<WebHookNotification>(s, options));
+                container.ReplaceSingleton(s => CreateAzureStorageQueue<MailMessage>(s, options));
+                container.ReplaceSingleton(s => CreateAzureStorageQueue<WorkItemData>(s, options, workItemTimeout: TimeSpan.FromHours(1)));
+            }
+        }
+
+        private static void RegisterStorage(IServiceCollection container, StorageOptions options) {
```

Comment:
Can you add support aws s3 (from foundatio.aws) please?

Reply:
Sure :)

Reply:
Added support for sqs and s3. it would be nice to have cloud watch support but that's a bit more than I want to chew off right now and could always be a future pr ;)